src/tokenizer.rs

```diff
@@ -258,11 +258,11 @@ pub struct Word {
 impl fmt::Display for Word {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self.quote_style {
-            Some(s) if s == '"' || s == '[' || s == '`' => {
-                write!(f, "{}{}{}", s, self.value, Word::matching_end_quote(s))
+            QuoteStyle::None => write!(f, "{}", self.value),
+            QuoteStyle::SingleQuote | QuoteStyle::DoubleQuote | QuoteStyle::BackTick => {
```

Comment:
I wonder if it would make sense to put the logic for printing quotes into `QuoteStyle` rather than partially there are partially here. It would make the `""` returned by `SquareBrack` easier to understand I think